protocol/message_persistence.go

```diff
@@ -488,6 +490,72 @@ func (db sqlitePersistence) MessageByChatID(chatID string, currCursor string, li
 	return result, newCursor, nil
 }
 
+// EmojiReactionsByChatID returns the emoji reactions for the queried messages, up to a maximum of 100, as it's a potentially unbound number.
+// NOTE: This is not completely accurate, as the messages in the database might have change since the last call to `MessageByChatID`.
+func (db sqlitePersistence) EmojiReactionsByChatID(chatID string, currCursor string, limit int) ([]*EmojiReaction, error) {
+	cursorWhere := ""
+	if currCursor != "" {
+		cursorWhere = "AND substr('0000000000000000000000000000000000000000000000000000000000000000' || m.clock_value, -64, 64) || m.id <= ?"
+	}
+	args := []interface{}{chatID, chatID}
+	if currCursor != "" {
+		args = append(args, currCursor)
+	}
+	args = append(args, limit)
+	// NOTE: We match against local_chat_id for security reasons.
+	// As a user could potentially send an emoji reaction for a one to
+	// one/group chat that has no access to.
+	// nolint: gosec
+	query := fmt.Sprintf(`
+			SELECT
+			    e.clock_value,
+			    e.source,
+			    e.emoji_id,
+			    e.message_id,
+			    e.chat_id,
+			    e.local_chat_id,
+			    e.retracted
+			FROM
+				emoji_reactions e
+			WHERE NOT(e.retracted)
+			AND
+			e.local_chat_id = ?
+			AND
+			e.message_id IN
+			(SELECT id FROM user_messages m WHERE NOT(m.hide) AND m.local_chat_id = ? %s
+			ORDER BY substr('0000000000000000000000000000000000000000000000000000000000000000' || m.clock_value, -64, 64) || m.id DESC LIMIT ?)
+			LIMIT 100
+		`, cursorWhere)
+
+	rows, err := db.db.Query(
+		query,
+		args...,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var result []*EmojiReaction
+	for rows.Next() {
+		var emojiReaction EmojiReaction
+		err := rows.Scan(&emojiReaction.Clock,
+			&emojiReaction.From,
+			&emojiReaction.Type,
+			&emojiReaction.MessageId,
+			&emojiReaction.ChatId,
+			&emojiReaction.LocalChatID,
+			&emojiReaction.Retracted)
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, &emojiReaction)
```

Comment:
Returning the whole list seems like a real overkill. This will get very heavy in threads taht have loads of reactions on loads of messages. A more sensible response would just include a count and a `bool` telling you if you are in the list.

Might be better to do it in a separate PR though.